R/get_fund_data.R

```diff
@@ -12,35 +14,51 @@
 
 get_fund_data <- function(
   conn,
-  data_timestamp
+  data_timestamp,
+  data_timestamp_lookback
 ) {
   # get the fund holdings and the holdings' reported market value ------------
 
   logger::log_debug("Extracting financial info from database.")
   logger::log_info("using data timestamp: ", data_timestamp)
+  logger::log_debug("Looking back in data to: ", data_timestamp_lookback)
 
   logger::log_trace(
     "Accessing historical fund holdings - security level. ",
     "Filtering to date: ", data_timestamp
   )
   fund_security <- dplyr::tbl(conn, "own_v5_own_fund_detail") %>%
-    dplyr::filter(.data[["report_date"]] == .env[["data_timestamp"]]) %>%
+    dplyr::filter(.data[["report_date"]] <= .env[["data_timestamp"]]) %>%
+    dplyr::filter(
+      .data[["report_date"]] >= .env[["data_timestamp_lookback"]]
+    ) %>%
+    dplyr::group_by(.data$factset_fund_id) %>%
+    dplyr::filter(.data$report_date == max(.data$report_date, na.rm = TRUE)) %>%
+    dplyr::ungroup() %>%
```

Comment:
nit: `dplyr::filter()` and friends have a new-ish `by`/`.by` argument that can be used to avoid the `group_by(); filter(); ungroup()` loop.

Reply:
also, you usually prefer...
```suggestion
    dplyr::group_by(.data[["factset_fund_id"]]) %>%
    dplyr::filter(.data[["report_date"]] == max(.data[["report_date"]], na.rm = TRUE)) %>%
    dplyr::ungroup() %>%
```

Reply:
I do! Thanks for catching.

Reply:
> nit: `dplyr::filter()` and friends have a new-ish `by`/`.by` argument that can be used to avoid the `group_by(); filter(); ungroup()` loop.

#58